wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/model/impl/WorkspaceRuntimeImpl.java

```diff
@@ -94,6 +94,10 @@ public void setMachines(List<MachineImpl> machines) {
         this.machines = machines;
     }
 
+    public void addMachine(MachineImpl machine) {
+        this.machines.add(machine);
```

Comment:
Can you please rework the [code in constructor](https://github.com/eclipse/che/blob/6ba34531215c09e98cb470937bdb32edaaff8f54/wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/model/impl/WorkspaceRuntimeImpl.java#L50-L52) to be npe safe.
```java
if (machines != null) {
    this.machines = machines.stream()
                            .map(MachineImpl::new)
                            .collect(toList());
}
```

and then reuse the getter `getMachines().add(machine)`, in this way the implementation becomes fully npe safe .